### PR TITLE
ppc64le port.

### PR DESCRIPTION
**Pull request by asowani**
This file contains changes required to build and use it on ppc64le platform.

No.

### CODE REVIEW

File: test/IRGen/big_types_corner_cases.sil [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // RUN: %target-swift-frontend -assume-parsing-unqualified-ownership-sil -I %S/Inputs/abi %s -emit-ir -enable-large-loadable-types | %FileCheck %s
 
 // UNSUPPORTED: resilient_stdlib
+// UNSUPPORTED: CPU=powerpc64le
```
Comment (jrose-apple) [inline test/IRGen/big_types_corner_cases.sil]: This shouldn't be necessary; the test is already limited to x86_64.

File: test/IRGen/big_types_corner_cases.swift [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // RUN: %target-swift-frontend -assume-parsing-unqualified-ownership-sil -enable-large-loadable-types %s -emit-ir | %FileCheck %s --check-prefix=CHECK --check-prefix=CHECK-%target-ptrsize
 // REQUIRES: optimized_stdlib
 // UNSUPPORTED: resilient_stdlib
+// UNSUPPORTED: CPU=powerpc64le
```
Comment (compnerd) [inline test/IRGen/big_types_corner_cases.swift]: Why is this unsupported on PPC64le?  Seems like its just disabling the test until it is fixed?  Can you file a SR for this?

Comment (asowani) [reply test/IRGen/big_types_corner_cases.swift]: @compnerd I think it's a good suggestion to mark big_types_corner_cases.swift as unsupported on ppc64le till it is analysed and fixed. I can certainly file a SR for this. However I couldn't find procedure to file SR in the contribution documents (https://github.com/apple/swift and https://swift.org/contributing/). Could you please direct me to a correct procedure to file SR? Thanks!

Comment (compnerd) [reply test/IRGen/big_types_corner_cases.swift]: @asowani the issues are tracked at http://bugs.swift.org ... you should be able to register and then file the SR there.

Comment (asowani) [reply test/IRGen/big_types_corner_cases.swift]: @compnerd Thanks for the pointer. I have created a new SR at https://bugs.swift.org/browse/SR-6953

Comment (asowani) [reply test/IRGen/big_types_corner_cases.swift]: @compnerd I have reverted changes in test/IRGen/big_types_corner_cases.sil as they were redundant. With this, is it possible to proceed with merging code changes in remaining files? Thanks!

**Comment by asowani** [comment]
@compnerd @jrose-apple Please let me know if anything needs to be done from my side (apart from the SR) to get these changes merged with master. Thanks!

**Comment by jrose-apple** [comment]
As these are just test case changes, I'm good with it if @compnerd has no further objections.

@swift-ci Please smoke test

**Review by asowani — COMMENTED** [review]
test/IRGen/clang_inline_opt.swift fails on ppc64le, hence it is now disabled on ppc64le. Expected value includes {{(zeroext)?}} on ppc64le.

**Comment by asowani** [comment]
@swift-ci Please smoke test

**Comment by asowani** [comment]
@compnerd @jrose-apple Any idea when this PR will be merged? Thanks!

**Comment by jrose-apple** [comment]
My bad. I was hoping Saleem would come back, but it seems like he's been busy over the last few months. I think you should just add the `{{(zeroext)? }}` pattern to the FileCheck line; no need to mark the whole test unsupported.

**Comment by sarveshtamba** [comment]
Following is the current status of the tests which are failing so far on **PowerPC64LE**:-
4 warning(s) in tests.
Testing Time: 1774.53s
********************
Failing Tests (19):
Swift(linux-powerpc64le) :: ClangImporter/cfuncs_parse.swift
Swift(linux-powerpc64le) :: Driver/sanitizers.swift
Swift(linux-powerpc64le) :: Driver/fuzzer.swift
Swift(linux-powerpc64le) :: IRGen/class_resilience_objc.swift
Swift(linux-powerpc64le) :: IRGen/big_types_corner_cases.swift
Swift(linux-powerpc64le) :: IRGen/clang_inline_opt.swift
Swift(linux-powerpc64le) :: IRGen/errors.sil
Swift(linux-powerpc64le) :: IRGen/objc_simd.sil
Swift(linux-powerpc64le) :: IRGen/objc_properties_jit.swift
Swift(linux-powerpc64le) :: IRGen/pic.swift
Swift(linux-powerpc64le) :: Index/Store/output-failure.swift
Swift(linux-powerpc64le) :: Interpreter/generic_casts.swift
Swift(linux-powerpc64le) :: ParseableInterface/ModuleCache/force-module-loading-mode-archs.swift
Swift(linux-powerpc64le) :: ParseableInterface/ModuleCache/force-module-loading-mode-framework.swift
Swift(linux-powerpc64le) :: ParseableInterface/ModuleCache/force-module-loading-mode.swift
Swift(linux-powerpc64le) :: Runtime/linux-fatal-backtrace.swift
Swift(linux-powerpc64le) :: Sanitizers/tsan.swift
Swift(linux-powerpc64le) :: remote-run/upload-stderr.test-sh
Swift(linux-powerpc64le) :: Sanitizers/witness_table_lookup.swift

Expected Passes : 10116
Expected Failures : 50
Unsupported Tests : 1341
Unexpected Failures: 19
Attached are the failure details of each of the failing tests.
[swift-test-failures-ppc64le.txt](https://github.com/apple/swift/files/2653074/swift-test-failures-ppc64le.txt)

**Comment by jrose-apple** [comment]
The PR was pending changes from Atul (the better solution for clang_inline_opt), and we've got test conflicts now. You're welcome to pick it up.

**Comment by sarveshtamba** [comment]
@jrose-apple sure, however can anyone confirm if these test cases are supported for PowerPC64LE?

**Comment by jrose-apple** [comment]
Atul added sensible CHECK lines to two of them, and our comments were about the third one. You should probably assume all test cases should be supported for PowerPC64LE if they're not already excluded by something like "CPU=x86_64". There'll probably be a few exceptions to that but in general I don't think we depend on many architectural details of x86_64 in our test cases.

**Comment by sarveshtamba** [comment]
Hi @jrose-apple, raised PR #21541 .  As suggested above, I think it's a good suggestion to mark failing tests as unsupported on ppc64le till it is analysed and fixed.
Filed SR - https://bugs.swift.org/browse/SR-9413

**Comment by CodaFi** [comment]
There hasn't been movement here in a year.  For anybody that wants to pick this back up, please refer to SR-9413.  Closing this out. Thanks @asowani.  This patch will be very valuable to a future contributor.

**Pull request closed without merging**